src/View/Helper/UrlHelper.php

```diff
@@ -16,15 +16,62 @@
  */
 namespace Cake\View\Helper;
 
+use Cake\Core\App;
+use Cake\Core\Exception\CakeException;
 use Cake\Routing\Asset;
 use Cake\Routing\Router;
 use Cake\View\Helper;
+use Cake\View\View;
 
 /**
  * UrlHelper class for generating URLs.
  */
 class UrlHelper extends Helper
 {
+    /**
+     * Default config for this class
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'engine' => Asset::class,
```

Comment:
`engine` key sounds weird in this case. Something like `assetUrlClassName` would be more appropriate.

Reply:
I thought the same. I had it as `assetEngine`, but since most of the helper is based on `Asset` class I left it as `engine` to keep more consistent with the other helpers.
I can still change if prefer to keep as `assetUrlClassName`

Reply:
Addressed it.